src/exports.js

```diff
@@ -39,45 +40,48 @@ var sqlite3_value_double;
-sqlite3_bind_blob = Module.cwrap('sqlite3_bind_blob', 'number', ['number', 'number', 'number', 'number', 'number']);
-sqlite3_bind_double = Module.cwrap('sqlite3_bind_double', 'number', ['number', 'number', 'number']);
-sqlite3_bind_int = Module.cwrap('sqlite3_bind_int', 'number', ['number', 'number', 'number']);
-sqlite3_bind_parameter_index = Module.cwrap('sqlite3_bind_parameter_index', 'number', ['number', 'string']);
-sqlite3_step = Module.cwrap('sqlite3_step', 'number', ['number']);
-sqlite3_errmsg = Module.cwrap('sqlite3_errmsg', 'string', ['number']);
-sqlite3_data_count = Module.cwrap('sqlite3_data_count', 'number', ['number']);
-sqlite3_column_double = Module.cwrap('sqlite3_column_double', 'number', ['number', 'number']);
-sqlite3_column_text = Module.cwrap('sqlite3_column_text', 'string', ['number', 'number']);
-sqlite3_column_blob = Module.cwrap('sqlite3_column_blob', 'number', ['number', 'number']);
-sqlite3_column_bytes = Module.cwrap('sqlite3_column_bytes', 'number', ['number', 'number']);
-sqlite3_column_type = Module.cwrap('sqlite3_column_type', 'number', ['number', 'number']);
-sqlite3_column_name = Module.cwrap('sqlite3_column_name', 'string', ['number', 'number']);
-sqlite3_reset = Module.cwrap('sqlite3_reset', 'number', ['number']);
-sqlite3_clear_bindings = Module.cwrap('sqlite3_clear_bindings', 'number', ['number']);
-sqlite3_finalize = Module.cwrap('sqlite3_finalize', 'number', ['number']);
-sqlite3_create_function_v2 = Module.cwrap('sqlite3_create_function_v2', 'number', ['number', 'string', 'number', 'number', 'number', 'number', 'number', 'number', 'number']);
-sqlite3_value_type = Module.cwrap('sqlite3_value_type', 'number', ['number']);
-sqlite3_value_bytes = Module.cwrap('sqlite3_value_bytes', 'number', ['number']);
-sqlite3_value_text = Module.cwrap('sqlite3_value_text', 'string', ['number']);
-sqlite3_value_int = Module.cwrap('sqlite3_value_int', 'number', ['number']);
-sqlite3_value_blob = Module.cwrap('sqlite3_value_blob', 'number', ['number']);
-sqlite3_value_double = Module.cwrap('sqlite3_value_double', 'number', ['number']);
-sqlite3_result_double = Module.cwrap('sqlite3_result_double', '', ['number', 'number']);
-sqlite3_result_null = Module.cwrap('sqlite3_result_null', '', ['number']);
-sqlite3_result_text = Module.cwrap('sqlite3_result_text', '', ['number', 'string', 'number', 'number']);
-sqlite3_result_blob = Module.cwrap('sqlite3_result_blob', '', ['number', 'number', 'number', 'number']);
-sqlite3_result_int = Module.cwrap('sqlite3_result_int', '', ['number', 'number']);
-sqlite3_result_int64 = Module.cwrap('sqlite3_result_int64', '', ['number', 'number']);
-sqlite3_result_error = Module.cwrap('sqlite3_result_error', '', ['number', 'string', 'number']);
-RegisterExtensionFunctions = Module.cwrap('RegisterExtensionFunctions', 'number', ['number']);
+// bug-workaround - emcc has issue minifying Module.cwrap,
+// so use Module["cwrap"] instead
+cwrap = Module['cwrap'];
```

Comment:
This is not a bug, this is a documented feature of the minifier. `object.property` can be renamed to anything, whereas `object['property']` enforces the use of the litteral `property` as the property's name.

https://developers.google.com/closure/compiler/docs/compilation_levels#advanced_optimizations

Reply:
sounds like bug to me.  there's no api/interface scenario where user would ever want (closure's) minifier to rename user-assigned properties -- and javascript is primarily used as tool for api/interface scenarios.

is there a way to disable closure's `ADVANCED_OPTIMIZATIONS` for everything except asm-related tasks?  i'm sorta skeptical of its performance benefits in normal, human-readable .js files.

Reply:
i've decided to use a modified version of jslint to resolve this issue (vanilla-jslint would forbid `object["property"]` if property is alphanumeric_and_underscore word).

```diff
/*
 * modified jslint.js
 */
-const rx_identifier = /^([a-zA-Z_$][a-zA-Z0-9_$]*)$/;
+// hack-jslint - ignore warning subscript_a
+const rx_identifier = /$^/m;
```